### PR TITLE
Zserio 2.14.0

### PR DESCRIPTION
**Pull request by Mi-La**
Specify library name and version:  **zserio/2.14.0**

Add new Zserio release 2.14.0.

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by Mi-La** [comment]
Hi! The only bigger change is that we added `zserio_compiler.cmake` script to our GitHub release and thus it will not be needed to use the one from conan-center-index repository anymore.
However it's still needed for zserio-2.13.0. I didn't know how to do it nicer than just to use `if`-s in the `conanfile.py`.
Any hints are very welcome ;-).

**Comment by Mi-La** [comment]
@uilianries Hello, just wondering what the status of this pull request? Do I need to do something more? Thanks.

**Review by RubenRBS — COMMENTED** [review]
Hi @Mi-La sorry for the delay, I misstagged the PR internally so did not get a reminder to circle back to this, sorry!
I'll review this alongside @uilianries tomorrow, but for now I have a minor question

File: recipes/zserio/all/conanfile.py
Comment (Mi-La) [reply recipes/zserio/all/conanfile.py:80]: Thanks for the quick response! There were some compilers in the pipelines which don't have C++11 enabled by default and the build didn't work - see build logs for [MacOS Clang 13](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/23846/2-macos-m1-clang/zserio/2.14.0//summary.json) and [gcc 5](https://c3i.jfrog.io/c3i/misc/summary.html?json=https://c3i.jfrog.io/c3i/misc/logs/pr/23846/1-linux-gcc-legacy/zserio/2.14.0//summary.json).
I just didn't know how to fix it and I've found similar solution in another packages.

The only variant I was thinking about was to add:
```python
if not valid_min_cppstd(self, self._min_cppstd):
    tc.variables["CMAKE_CXX_STANDARD"] = self._min_cppstd
```
which would probably allow to set higher C++ standard. However the Zserio C++ generator states that it's C++11, so I decided to force the standard to C++11. But I would probably accept any solution which will ensure that at least C++11 will be used.

Comment (RubenRBS) [reply recipes/zserio/all/conanfile.py:80]: Oh, I see! This is one of the painpoints we hope to alleviate once we drop support for Conan 1 in the pipeline, but for now we need to take care of these situations.

If you check the "Contributors are allowed to push to this branch" in the righthand pane in the PR, I'll be able to directly push a fix for you for this case :)

Comment (Mi-La) [reply recipes/zserio/all/conanfile.py:80]: > Contributors are allowed to push to this branch

I don't see such an option, the PR is created from a fork, maybe it's the reason? I think you will have to create a PR on my branch? https://github.com/ndsev/zserio-conan-center-index/tree/zserio-2.14.0

Comment (RubenRBS) [reply recipes/zserio/all/conanfile.py:80]: Weird, I had supposed that https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests would show up at all times!

Comment (Mi-La) [reply recipes/zserio/all/conanfile.py:80]: Maybe it's related to this [issue](https://github.com/isaacs/github/issues/1681)? Also in a [discussion](https://github.com/orgs/community/discussions/5634).
image

Comment (RubenRBS) [reply recipes/zserio/all/conanfile.py:80]: TIL, thanks for investigating it :) I'll open a pr on your fork then, sorry for the noise!

Comment (uilianries) [reply recipes/zserio/all/conanfile.py:80]: > However the Zserio C++ generator states that it's C++11, so I decided to force the standard to C++11.

I would suggest opening an issue to the upstream asking to fix it. The upstream would expect a modern compiler, that's not the first time we such case. It could be fixed by including `target_compiler_features(<target> PUBLIC cxx_std_11)` in their CMake file.

Comment (Mi-La) [reply recipes/zserio/all/conanfile.py:80]: > would impose minimal C++11 standard as required

Aha, I didn't know that, thanks, we will investigate it the related issue... However the Zserio release 2.14.0 is already published...

Comment (uilianries) [inline recipes/zserio/all/conanfile.py:80]:
```suggestion
        tc.variables["CMAKE_CXX_STANDARD"] = self.settings.compiler.get_safe("cppstd", str(self._min_cppstd)).replace("gnu", "")
```
It will respect settings.compiler.cppstd in case defined, otherwise, will define 11 (min_cppstd)

File: recipes/zserio/all/conanfile.py
Comment (uilianries) [inline recipes/zserio/all/conanfile.py:35]: C++11 is supported since GCC 4.7 (not all features), the GCC 7 supports C++14 as well. Can we drop the minimal compiler version required, or does it require C++14?

File: recipes/zserio/all/conanfile.py
Comment (uilianries) [inline recipes/zserio/all/conanfile.py:89]: It could be downloaded in source(). The compiler is different because is a binary, but cmake file we download directly to sources.

File: recipes/zserio/all/conanfile.py
Comment (Mi-La) [reply recipes/zserio/all/conanfile.py:35]: Hm .. I probably forgot to allow gcc 5 after the fix with minimum C++ standard version. However our [documentation](https://github.com/ndsev/zserio/tree/v2.13.0/compiler/extensions/cpp) for first Zserio version in Conan states that it supports gcc 5.4. Thus I would like to change it at least to `"gcc" :  "5"`.

Or is it supposed that it is user's problem when older gcc version that the version supported by library is used? The other question is what should I do if Zserio 2.14.0 build fails with gcc 5 in your pipelines (note that we now support [gcc 7.4.0+](https://github.com/ndsev/zserio/tree/v2.13.0/compiler/extensions/cpp))

File: recipes/zserio/all/conanfile.py
Comment (Mi-La) [reply recipes/zserio/all/conanfile.py:89]: I don't understand. The script is released as part of the [compiler zip release](https://github.com/ndsev/zserio/releases/download/v2.14.0/zserio-2.14.0-bin.zip) since it is an utility for compiler users -> to be able to call the compiler from any CMake project.

Should I have an extra source like [this](https://raw.githubusercontent.com/ndsev/zserio/v2.14.0/cmake/zserio_compiler.cmake)? And use it only from Zserio version 2.14.0 since it wasn't in the repository at all in Zserio 2.13.0?

File: recipes/zserio/all/conanfile.py
Comment (Mi-La) [reply recipes/zserio/all/conanfile.py]: changed to gcc 5 for now

File: recipes/zserio/all/conanfile.py
Comment (uilianries) [reply recipes/zserio/all/conanfile.py:89]: Okay, makes sense.

**Review by uilianries — DISMISSED** [review]

**Comment by uilianries** [comment]
@Mi-La Sorry the review delay, but when having changes in the release we really need be careful reviewing the recipe to avoid future patches and bugs. The whole process delayed more than could be, yes, but keep in mind we have more 600 PRs waiting for review, even other authors publishing their releases too, so we are doing our best to keep all PRs reviewed as much as possible.

**Comment by Mi-La** [comment]
@uilianries I absolutely understand that it's complicated to review all the changes manually, moreover when users have almost unlimited possibilities how to write the recipe, which is the biggest problem I think.

I hope that next time I'll have almost no changes, just add the links to the new release and it will simply work :-).

Maybe in future we could completely delete the Zserio 2.13 release from conan and simplify the recipe again? Would it be acceptable?

**Comment by uilianries** [comment]
> Maybe in future we could completely delete the Zserio 2.13 release from conan and simplify the recipe again? Would it be acceptable?

Totally! Thank you for your understanding 😄

**Comment by RubenRBS** [comment]
> I hope that next time I'll have almost no changes, just add the links to the new release and it will simply work :-).

The community has provided tools to make this even faster for you, check https://github.com/conan-io/conan-center-index/issues/3470 for one such case :)

**Comment by Mi-La** [comment]
@RubenRBS Could you please approve the PR, or shall I still do some changes?

File: recipes/zserio/all/conanfile.py
Comment (RubenRBS) [inline recipes/zserio/all/conanfile.py:80]:
```suggestion
if not self.settings.get_safe("compiler.cppstd"): 
           tc.variables["CMAKE_CXX_STANDARD"] = str(self._min_cppstd)
```

**Review by RubenRBS — COMMENTED** [review]
Hi! This PR is most of the way there, only a minor thing left missing from me, but this will get merged today :)

**Review by uilianries — APPROVED** [review]

**Review by RubenRBS — APPROVED** [review]

**Pull request merged**
